src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -270,6 +256,17 @@ public Mono<ShippingInstructionResponseTO> createShippingInstruction(
               return Mono.just(si);
             })
         .flatMap(si -> createShipmentEvent(si).thenReturn(si))
+        .flatMap(
+            si -> {
+              si.setValidUntil(OffsetDateTime.now());
+              return shippingInstructionRepository.save(si);
+            })
+        .flatMap(
+            si -> {
+              si.setId(null);
+              si.setValidUntil(null);
+              return shippingInstructionRepository.save(si);
+            })
```

Comment:
I think the related shipment event should be associated with the _new_ object (rather than the old).  Otherwise the references will be wrong (i.e., use the old references rather than the new one) in the related event.

Reply:
Ah, I think you're right. Moved it to after the creation of the new shipping instruction.